packages/template-retail-react-app/app/constants.js

```diff
@@ -201,3 +201,29 @@ export const STORE_LOCATOR_DISTANCE = 100
 export const STORE_LOCATOR_NUM_STORES_PER_LOAD = 10
 export const STORE_LOCATOR_DISTANCE_UNIT = 'km'
 export const STORE_LOCATOR_IS_ENABLED = true
+
+// Constants for shopper context
+export const SHOPPER_CONTEXT_FIELD_TYPES = {
+    INT: 'int',
+    DOUBLE: 'double',
+    ARRAY: 'array'
+}
+export const SHOPPER_CONTEXT_SEARCH_PARAM_TO_API_FIELD_MAPPING = {
+    sourceCode: {apiField: 'sourceCode'},
+    effectiveDateTime: {apiField: 'effectiveDateTime'},
+    customerGroupIds: {apiField: 'customerGroupIds', type: SHOPPER_CONTEXT_FIELD_TYPES.ARRAY},
+    clientIp: {apiField: 'clientIp'}
+}
+export const SHOPPER_CONTEXT_GEOLOCATION_SEARCH_PARAM_TO_API_FIELD_MAPPING = {
+    city: {apiField: 'city'},
+    country: {apiField: 'country'},
+    countryCode: {apiField: 'countryCode'},
+    latitude: {apiField: 'latitude', type: SHOPPER_CONTEXT_FIELD_TYPES.DOUBLE},
+    longitude: {apiField: 'longitude', type: SHOPPER_CONTEXT_FIELD_TYPES.DOUBLE},
+    metroCode: {apiField: 'metroCode'},
+    postalCode: {apiField: 'postalCode'},
+    region: {apiField: 'region'},
+    regionCode: {apiField: 'regionCode'}
```

Comment:
@alexvuong i've updated the constants here to take an object so we can specify the type. 

Now if users want to change the name of the query param they would just update the key